[CI] ActionErrorIntegrationTests testErrorInAction failing

**Build scan:**
https://gradle-enterprise.elastic.co/s/vmkb5th5kwiy4/tests/:x-pack:plugin:watcher:internalClusterTest/org.elasticsearch.xpack.watcher.actions.ActionErrorIntegrationTests/testErrorInAction


**Reproduction line:**
```
./gradlew ':x-pack:plugin:watcher:internalClusterTest' --tests "org.elasticsearch.xpack.watcher.actions.ActionErrorIntegrationTests.testErrorInAction" -Dtests.seed=D8614AB7D6393F3D -Dtests.locale=de-LU -Dtests.timezone=Europe/Bucharest -Druntime.java=18
```

**Applicable branches:**
main

**Reproduces locally?:**
Didn't try

**Failure history:**
[Failure dashboard for `org.elasticsearch.xpack.watcher.actions.ActionErrorIntegrationTests#testErrorInAction`](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('testErrorInAction'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.watcher.actions.ActionErrorIntegrationTests'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))


**Failure excerpt:**
```
org.elasticsearch.action.search.SearchPhaseExecutionException: all shards failed

  at __randomizedtesting.SeedInfo.seed([D8614AB7D6393F3D:7576EB4B291DC618]:0)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseFailure(AbstractSearchAsyncAction.java:731)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.executeNextPhase(AbstractSearchAsyncAction.java:422)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onPhaseDone(AbstractSearchAsyncAction.java:763)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.onShardFailure(AbstractSearchAsyncAction.java:516)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.performPhaseOnShard(AbstractSearchAsyncAction.java:330)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.run(AbstractSearchAsyncAction.java:266)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.executePhase(AbstractSearchAsyncAction.java:473)
  at org.elasticsearch.action.search.AbstractSearchAsyncAction.start(AbstractSearchAsyncAction.java:221)
  at org.elasticsearch.action.search.TransportSearchAction.executeSearch(TransportSearchAction.java:1123)
  at org.elasticsearch.action.search.TransportSearchAction.executeLocalSearch(TransportSearchAction.java:892)
  at org.elasticsearch.action.search.TransportSearchAction.lambda$executeRequest$9(TransportSearchAction.java:321)
  at org.elasticsearch.action.ActionListenerImplementations$ResponseWrappingActionListener.onResponse(ActionListenerImplementations.java:236)
  at org.elasticsearch.index.query.Rewriteable.rewriteAndFetch(Rewriteable.java:109)
  at org.elasticsearch.index.query.Rewriteable.rewriteAndFetch(Rewriteable.java:77)
  at org.elasticsearch.action.search.TransportSearchAction.executeRequest(TransportSearchAction.java:433)
  at org.elasticsearch.action.search.TransportSearchAction.doExecute(TransportSearchAction.java:288)
  at org.elasticsearch.action.search.TransportSearchAction.doExecute(TransportSearchAction.java:111)
  at org.elasticsearch.action.support.TransportAction$RequestFilterChain.proceed(TransportAction.java:87)
  at org.elasticsearch.action.support.TransportAction.execute(TransportAction.java:62)
  at org.elasticsearch.tasks.TaskManager.registerAndExecute(TaskManager.java:197)
  at org.elasticsearch.client.internal.node.NodeClient.executeLocally(NodeClient.java:113)
  at org.elasticsearch.client.internal.node.NodeClient.doExecute(NodeClient.java:91)
  at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:379)
  at org.elasticsearch.client.internal.support.AbstractClient.execute(AbstractClient.java:365)
  at org.elasticsearch.action.ActionRequestBuilder.execute(ActionRequestBuilder.java:34)
  at org.elasticsearch.action.ActionRequestBuilder.get(ActionRequestBuilder.java:41)
  at org.elasticsearch.xpack.watcher.test.AbstractWatcherIntegrationTestCase.docCount(AbstractWatcherIntegrationTestCase.java:319)
  at org.elasticsearch.xpack.watcher.test.AbstractWatcherIntegrationTestCase.watchRecordCount(AbstractWatcherIntegrationTestCase.java:314)
  at org.elasticsearch.xpack.watcher.actions.ActionErrorIntegrationTests.lambda$testErrorInAction$0(ActionErrorIntegrationTests.java:55)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1190)
  at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:1175)
  at org.elasticsearch.xpack.watcher.actions.ActionErrorIntegrationTests.testErrorInAction(ActionErrorIntegrationTests.java:54)
  at jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
  at java.lang.reflect.Method.invoke(Method.java:577)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.invoke(RandomizedRunner.java:1758)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$8.evaluate(RandomizedRunner.java:946)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$9.evaluate(RandomizedRunner.java:982)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$10.evaluate(RandomizedRunner.java:996)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleSetupTeardownChained$1.evaluate(TestRuleSetupTeardownChained.java:48)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleThreadAndTestName$1.evaluate(TestRuleThreadAndTestName.java:45)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.forkTimeoutingTask(ThreadLeakControl.java:843)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$3.evaluate(ThreadLeakControl.java:490)
  at com.carrotsearch.randomizedtesting.RandomizedRunner.runSingleTest(RandomizedRunner.java:955)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$5.evaluate(RandomizedRunner.java:840)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$6.evaluate(RandomizedRunner.java:891)
  at com.carrotsearch.randomizedtesting.RandomizedRunner$7.evaluate(RandomizedRunner.java:902)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleStoreClassName$1.evaluate(TestRuleStoreClassName.java:38)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.NoShadowingOrOverridesOnMethodsRule$1.evaluate(NoShadowingOrOverridesOnMethodsRule.java:40)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at org.apache.lucene.tests.util.TestRuleAssertionsRequired$1.evaluate(TestRuleAssertionsRequired.java:53)
  at org.apache.lucene.tests.util.AbstractBeforeAfterRule$1.evaluate(AbstractBeforeAfterRule.java:43)
  at org.apache.lucene.tests.util.TestRuleMarkFailure$1.evaluate(TestRuleMarkFailure.java:44)
  at org.apache.lucene.tests.util.TestRuleIgnoreAfterMaxFailures$1.evaluate(TestRuleIgnoreAfterMaxFailures.java:60)
  at org.apache.lucene.tests.util.TestRuleIgnoreTestSuites$1.evaluate(TestRuleIgnoreTestSuites.java:47)
  at com.carrotsearch.randomizedtesting.rules.StatementAdapter.evaluate(StatementAdapter.java:36)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl$StatementRunner.run(ThreadLeakControl.java:390)
  at com.carrotsearch.randomizedtesting.ThreadLeakControl.lambda$forkTimeoutingTask$0(ThreadLeakControl.java:850)
  at java.lang.Thread.run(Thread.java:833)

  Caused by: org.elasticsearch.action.NoShardAvailableActionException: (No message provided)


```

--- a/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/actions/ActionErrorIntegrationTests.java
+++ b/x-pack/plugin/watcher/src/internalClusterTest/java/org/elasticsearch/xpack/watcher/actions/ActionErrorIntegrationTests.java
@@ -6,6 +6,7 @@
  */
 package org.elasticsearch.xpack.watcher.actions;
 
+import org.elasticsearch.ElasticsearchException;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.index.query.QueryBuilders;
@@ -52,12 +53,20 @@ public class ActionErrorIntegrationTests extends AbstractWatcherIntegrationTestC
 
         // there should be a single history record with a failure status for the action:
         assertBusy(() -> {
-            long count = watchRecordCount(
-                QueryBuilders.boolQuery()
-                    .must(termsQuery("result.actions.id", "_action"))
-                    .must(termsQuery("result.actions.status", "failure"))
-            );
-            assertThat(count, is(1L));
+            try {
+                long count = watchRecordCount(
+                    QueryBuilders.boolQuery()
+                        .must(termsQuery("result.actions.id", "_action"))
+                        .must(termsQuery("result.actions.status", "failure"))
+                );
+                assertThat(count, is(1L));
+            } catch (ElasticsearchException e) {
+                /*
+                 * Since the history is written asynchronously, it is possible that we try to query it after the history index is
+                 * created, but before the shards are allocated, which throws an exception.
+                 */
+                throw new AssertionError(e);
+            }
         });
 
         // now we'll trigger the watch again and make sure that it's not throttled and instead
@@ -71,12 +80,20 @@ public class ActionErrorIntegrationTests extends AbstractWatcherIntegrationTestC
 
         // there should be a single history record with a failure status for the action:
         assertBusy(() -> {
-            long count = watchRecordCount(
-                QueryBuilders.boolQuery()
-                    .must(termsQuery("result.actions.id", "_action"))
-                    .must(termsQuery("result.actions.status", "failure"))
-            );
-            assertThat(count, is(2L));
+            try {
+                long count = watchRecordCount(
+                    QueryBuilders.boolQuery()
+                        .must(termsQuery("result.actions.id", "_action"))
+                        .must(termsQuery("result.actions.status", "failure"))
+                );
+                assertThat(count, is(2L));
+            } catch (ElasticsearchException e) {
+                /*
+                 * Since the history is written asynchronously, it is possible that we try to query it after the history index is
+                 * created, but before the shards are allocated, which throws an exception.
+                 */
+                throw new AssertionError(e);
+            }
         });
 
         // now lets confirm that the ack status of the action is awaits_successful_execution

